core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -68,11 +67,11 @@ public class SharePartitionManager {
     private final ConcurrentLinkedQueue<ShareFetchPartitionData> fetchQueue;
     private final AtomicBoolean processFetchQueueLock;
 
-    public SharePartitionManager(ReplicaManager replicaManager, Time time, ShareSessionCache cache) {
+    public SharePartitionManager(ReplicaManager replicaManager, Time time, Map<SharePartitionKey, SharePartition> partitionCacheMap, ShareSessionCache cache) {
```

Comment:
@apoorvmittal10 do you think it makes sense that we add `partitionCacheMap` to the constructor of SharePartitionManager, when in reality, it has to be populated by fetch/acknowledge operations within the SharePartitionManager. I see it in the constructor call in `BrokerServer.scala`, it is always initialized as an empty map

Reply:
Maybe just making `partitionCacheMap` as Default might be better?

Reply:
The way to do this this adding an overloaded constructor which takes:
1. `SharePartitionManager(ReplicaManager replicaManager, Time time, Map<SharePartitionKey, SharePartition> partitionCacheMap, ShareSessionCache cache)` arguments. 
2. The old public constructor should not expose `Map<SharePartitionKey, SharePartition> partitionCacheMap` hence no need to change anywhere else in the code.
3. The new overloaded constructor should be of `default` visibility scope so it's exposed to tests.
4. The new overloaded constructor should have a comment on top `// Visible for testing`. You will see such instances in the current code as well.